transports/webtransport-websys/src/lib.rs

```diff
@@ -0,0 +1,21 @@
+//! Libp2p WebTransport built on [web-sys](https://rustwasm.github.io/wasm-bindgen/web-sys/index.html)
+//!
+//! This crate uses some unstable apis of web-sys so it requires
+//! `--cfg=web_sys_unstable_apis` to be activated, as [described
+//! in the `wasm-bindgen` guide](https://rustwasm.github.io/docs/wasm-bindgen/web-sys/unstable-apis.html)
+
+mod connection;
+mod endpoint;
+mod error;
+mod fused_js_promise;
+mod stream;
+mod transport;
+mod utils;
+
+pub use self::connection::Connection;
+pub use self::error::Error;
+pub use self::stream::Stream;
+pub use self::transport::{Config, Transport};
+
+#[cfg(test)]
+wasm_bindgen_test::wasm_bindgen_test_configure!(run_in_browser);
```

Comment:
The test tha actually needs WASM currently is `utils::tests::check_js_typecasting` only. However I'm trying to figure out how I can do a connectivity test.

Because a server implementation is not possible with web-sys, I don't know how I can properly write a test for connectivity. The only idea that came to my mind, is to write an echo server in Go and spawn the executable before the tests run.

Will this be acceptable? If yes, in which directory should I add the code of Go echo server?

Reply:
Resolving here in favor of https://github.com/libp2p/rust-libp2p/pull/4015#issuecomment-1578431779.